pysat/utils/io.py

```diff
@@ -172,121 +167,532 @@ def filter_netcdf4_metadata(inst, mdata_dict, coltype, remove=False,
-
-                for i, dim in enumerate(dims[:-1]):
-                    # Don't need to go over last dimension value,
-                    # it covers number of columns (if a DataFrame)
-                    obj_dim_names.append(var)
-
-                # Set the base-level meta data
-                meta_dict['Depend_1'] = obj_dim_names[-1]
-
-                # Cycle through each of the sub-variable, updating metadata
-                for svar in subvars:
-                    # Get the correct location of the sub-variable based on
-                    # the object type
-                    if is_frame:
-                        idx = inst[var].iloc[0][svar]
-                    else:
-                        idx = inst[var].iloc[0]
+                if labels.units in mdict[key]:
+                    mdict[key][labels.units] = ''
 
-                    # Attach the metadata
-                    _, scoltype, _ = inst._get_data_info(idx)
+                break
 
-                    smeta_dict = {'Depend_0': epoch_name,
-                                  'Depend_1': obj_dim_names[-1],
-                                  'Display_Type': 'Spectrogram',
-                                  'Format': inst._get_var_type_code(scoltype),
-                                  'Var_Type': 'data'}
-                    inst.meta[svar] = smeta_dict
+        # Remove any entries with label in `vals`
+        for val, lval in zip(vals, lower_vals):
+            if lval in lower_sub_keys:
+                for i, check_val in enumerate(lower_sub_keys):
+                    if check_val == lval:
+                        mdict[key].pop(sub_keys[i])
+
+    # Remove epoch metadata
+    if epoch_name != '':
+        if epoch_name in mdict:
+            mdict.pop(epoch_name)
+
+    return mdict
+
+
+def default_from_netcdf_translation_table(meta):
+    """Return metadata translation table with minimal netcdf requirements.
+
+    Parameters
+    ----------
+    meta : pysat.Meta
+        Meta instance to get appropriate default values for.
+
+    Returns
+    -------
+    dict
+        Keyed by self.labels with a list of strings to be used
+        when writing netcdf files.
+
+    Note
+    ----
+    The purpose of this function is to maintain default compatibility
+    with `meta.labels` and existing code that writes and reads netcdf
+    files via pysat while also changing the labels for metadata within
+    the file.
+
+    """
+
+    # Define a default translation with labels required by netCDF4.
+    trans_table = {'_FillValue': meta.labels.fill_val,
+                   'FillVal': meta.labels.fill_val,
+                   'fill': meta.labels.fill_val}
+
+    # We need to keep 'fill' for backwards compatibility. Unintended use
+    # of 'fill' in pysat generated files in at least v3.0.1.
+
+    return trans_table
+
+
+def default_to_netcdf_translation_table(inst):
+    """Return metadata translation table with minimal netcdf requirements.
```

Comment:
```suggestion
    """Create a metadata translation table with minimal netCDF requirements.
```

For the docstrings in this file, recommend using "Create" or "Extract", etc instead of "Return", since the latter is technically covered by the Returns section and the former give more insight into the innards of the function.